pkg/resourcemanager/mysql/mysqlhelper.go

```diff
@@ -88,31 +90,72 @@ func ConnectToSQLDBAsCurrentUser(
 	return db, err
 }
 
+// ExtractUserRoles extracts the roles the user has. The result is the set of roles the user has for the requested database
+func ExtractUserRoles(ctx context.Context, db *sql.DB, user string, database string) (map[string]struct{}, error) {
+	if err := helpers.FindBadChars(user); err != nil {
+		return nil, errors.Wrapf(err, "problem found with username")
+	}
+
+	// Note: This works because we only assign permissions at the DB level, not at the table, column, etc levels -- if we assigned
+	// permissions at more levels we would need to do something else here such as join multiple tables or
+	// parse SHOW GRANTS with a regex.
+	tsql := fmt.Sprintf("SELECT PRIVILEGE_TYPE from information_schema.SCHEMA_PRIVILEGES WHERE GRANTEE = '''%s''@''%%''' and TABLE_SCHEMA = ?", user)
+	rows, err := db.QueryContext(ctx, tsql, database)
```

Comment:
Grantee can be a parameter here, can't it? You'd still have to build the weird formatting with the quotes and @ - that's a bit easier if you don't need to sql-quote it.
```suggestion
	tsql := "select privilege_type from information_schema.schema_privileges where grantee = ? and table_schema= ?"
	formattedUser := fmt.Sprintf("'%s'@'%'", user)
	rows, err := db.QueryContext(ctx, tsql, formattedUser, database)
```

I tend to make queries all lowercase, not sure what the standard is - a lot of people prefer SQL keywords in uppercase.

Reply:
Yeah all of the existing queries we have are all uppercase for the keywords at least. I'll fix this